File: plugins/daysuntil.py
# -*- coding: utf-8 -*-
from datetime import date
TARGET_DATE = date(2019, 7, 4)


def main():
    global TARGET_DATE
    days_until = (TARGET_DATE - date.today()).days

    result = (
        f" <blue>i<reset> There {'are' if days_until != 1 else 'is'} "
        f"<blue>{str(days_until) if days_until >= 0 else '0'} "
        f"{'days' if days_until != 1 else 'day'}<reset> "
        f"until <green>{TARGET_DATE.strftime('%B %d, %Y')}."
    )

    print(result)


if __name__ == '__main__':
    main()

File: plugins/daysuntil.go
package plugins

import (
	"fmt"
	"strconv"
	"time"

	"github.com/fatih/color"
)

func PluginDaysUntil() {
	targetDate := time.Date(2025, 10, 14, 0, 0, 0, 0, time.Local)
	currentDate := time.Now()
	daysUntil := targetDate.Sub(time.Date(
        currentDate.Year(), currentDate.Month(), currentDate.Day(),
        0, 0, 0, 0, currentDate.Location(),
    ))

	fmt.Print(color.GreenString(" i ") + "There are ")
	fmt.Print(color.GreenString(strconv.FormatInt(int64(daysUntil.Hours()/24), 10)))
	fmt.Print(" days until " + color.GreenString(targetDate.Format("January 2, 2006")) + ".")
}
